docs/core/tutorials/netcore-hosting.md

```diff
@@ -0,0 +1,312 @@
+
+### Step 8 - Clean up
+Finally, the host should clean up after itself by unloading AppDomains, stopping the runtime, and releasing the `ICLRRuntimeHost2` reference.
+
+```C+++
+runtimeHost->UnloadAppDomain(domainId, true /* Wait until unload complete */);
+runtimeHost->Stop();
+runtimeHost->Release();
+``` 
+
+## About Hosting .NET Core on Unix
+.NET Core is a cross-platform product, running on Windows, Linux, and Mac operating systems. As native applications, though, hosts for different platforms will have some differences between them. The process described above of using `ICLRRuntimeHost2` to start the runtime, create an AppDomain, and execute managed code, should work on any supported operating system. However, the interfaces defined in mscoree.h can be cumbersome to work with on Unix platforms since mscoree makes many Win32 assumptions.
+
+To make hosting on Unix platforms easier, a set of more platform-neutral hosting API wrappers are available in [coreclrhost.h](https://github.com/dotnet/coreclr/blob/master/src/coreclr/hosts/inc/coreclrhost.h).
+
+An example of using coreclrhost.h (instead of mscoree.h directly) can be seen in the [UnixCoreRun host](https://github.com/dotnet/coreclr/tree/master/src/coreclr/hosts). The steps to use the APIs from coreclrhost.h to host the runtime are similar to the steps when using mscoree.h:
+
+1. Identify the managed code to execute (from command line parameters, for example). 
+2. Load the CoreCLR library.
+	1. `dlopen("./libcoreclr.so", RTLD_NOW | RTLD_LOCAL);` 
+3. Get function pointers to CoreCLR's `coreclr_initialize`, `coreclr_create_delegate`, `coreclr_execute_assembly`, and `coreclr_shutdown` functions using `dlsym`
+	1. `coreclr_initialize_ptr coreclr_initialize = (coreclr_initialize_ptr)dlsym(coreclrLib, "coreclr_initialize");`
+4. Setup AppDomain properties (such as the TPA list). This is the same as step 5 from the mscoree workflow, above.
+5. Use `coreclr_initialize` to start the runtime and create an AppDomain. This will also create a `hostHandle` pointer which will be used in future hosting calls.
+	1. Note that this function fills the roles of both steps 4 and 6 from the previous workflow. 
+6. Use either `coreclr_execute_assembly` or `coreclr_create_delegate` to execute managed code. These functions are analogous to mscoree's `ExecuteAssembly` and `CreateDelegate` functions from step 7 of the previous workflow.
+7. Use `coreclr_shutdown` to unload the AppDomain and shut down the runtime. 
+
+## Conclusion
+Once your host is built, it can be tested by running it from the command line and passing any arguments (like the managed app to run) the host expects. When specifying the .NET Core app for the host to run, be sure to use the .dll that is produced by `dotnet build`. Executables produced by `dotnet build` are actually the default .NET Core host (so that the app can be launched directly from the command line in mainline scenarios); user's code is compiled into a dll of the same name. 
```

Comment:
As I recall, compile doesn't produce an exe; only publish does, and only for self-contained deployments.